File: output/llama3.1-8b_IO_1/avatar/Java/Python/atcoder_AGC027_B.py
import sys
class atcoder_AGC027_B:
    INF = sys.maxsize
    def solve(self):
        N, X = map(int, input().split())
        x = list(map(int, input().split()))
        xsum = [0] * (N + 1)
        for i in range(N):
            xsum[i+1] = xsum[i] + x[i]
        ans = X * N + 5 * sum(x)

        for i in range(1, N):
            cost = X * i + 5 * (sum(x[:N-i]) - sum(x[:i]))
            j, k = 5, N - i
            while k >= 0:
                if cost > ans:
                    break
                cost += j * (xsum[k] - xsum[max(k - i, 0)])
                j += 2
                k -= i
            ans = min(ans, cost)

        print(ans + N * X)
atcoder_AGC027_B().solve()

File: output/llama3.1-8b_IO_1/avatar/Python/Java/atcoder_AGC027_B.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class atcoder_AGC027_B {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader("input.txt"));
        String[] input = br.readLine().split("\\s+");
        int N = Integer.parseInt(input[0]);
        int X = Integer.parseInt(input[1]);
        int[] A = new int[N];
        for (int i = 2; i < input.length; i++) {
            A[i - 2] = Integer.parseInt(input[i]);
        }

        long[] S = new long[N];
        long sum = 0;
        for (int i = 0; i < N; i++) {
            sum += A[i];
            S[i] = sum;
        }

        long ans = Long.MAX_VALUE;
        for (int k = 1; k <= N / 2 + 1; k++) {
            int start = Math.max(0, N - 2 * k - 1);
            long E = k * X + 2 * 0;
            for (int j = start; j >= 0; j -= k) {
                E += S[j];
            }
            ans = Math.min(ans, E);
        }

        System.out.println(ans + N * X + 5 * S[S.length - 1]);
    }
}
